### PR TITLE
Show valid P-Chain address for a given Node ID during registration

### PR DESCRIPTION
**Pull request by emersoncloud**
During registration we had a lot of questions from people about which P-Chain and C-Chain addresses they should be using when creating the signed message. 

This PR adds a field for them to input their NodeID, and with a call to the glacier API we get the P-Chain address the user should use, and verify that it will work to register. 

image

### CODE REVIEW

File: src/components/Pages/Register/RegisterTop.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        </span>
+      </div>
+      <div className="flex w-full justify-center">
+        <div className="flex w-full flex-col gap-2 text-black pt-4 basis-[520px]">
+          <div className="flex w-full justify-center mb-2">
+            <div className="flex w-full justify-center mt-4 border-2 border-transparent focus-within:border-2 focus-within:border-black">
+              <input
+                value={nodeId}
+                onChange={handleChange}
+                className="text-md w-full border-2 border-primary-500 bg-black focus-within:outline-none bg-opacity-10 p-3 text-secondary-800"
```
Comment (jacktrusler) [inline src/components/Pages/Register/RegisterTop.tsx:100]: You don't need the `<div>` on line 95, you can delete that and add mb-2 to the div on line 96. Then you don't need width-full on the div that has `basis` or the `<div>` on line 96.

**Review by jacktrusler — CHANGES_REQUESTED** [review]
Everything looks good, some small changes. CSS can ligma.

File: src/components/Pages/Register/RegisterTop.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        setNodeIdError('NodeId not found');
+        return;
+      }
+      const pAddr = validators[0].potentialRewards?.rewardAddresses[0];
+      if (pAddr === null || pAddr === undefined) {
+        setNodeIdError('Unable to fetch P-Chain address');
+        return;
+      }
+      setPChainAddr(pAddr);
+      setContextVerified();
```
[the remark this answers is not in the document]
Comment (emersoncloud) [reply src/components/Pages/Register/RegisterTop.tsx]: Ya you right. I thought I was being clever by having the text in the function not randomly in the code, but I ended up just putting it in the HTML

File: src/components/Pages/Register/RegisterTop.tsx (hunk 1 above)
Comment (emersoncloud) [reply src/components/Pages/Register/RegisterTop.tsx]: 🖖

**Review by jacktrusler — APPROVED** [review]

**Comment by WillDosSantos** [comment]
@emersoncloud @jacktrusler 

Looking awesome! A couple small things I noticed here:
https://jumpshare.com/v/WJFA6fCkM8H7wvL8J6WY

**Pull request merged**